code/game/objects/items/tanks/jetpack.dm

```diff
@@ -55,6 +56,7 @@
 	icon_state = "[initial(icon_state)]-on"
 	ion_trail.start()
 	RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/move_react)
+	RegisterSignal(user, COMSIG_MOVABLE_PRE_MOVE, .proc/pre_move_react)

```

Comment:
pre move could result in no move happening at all, if you want side effects from a move you should be making used of moved

Reply:
pre move now only save old turf

Reply:
sure but the premove can be set off and no move happens, it may not be the old turf but actualy the current turf when everything is done

Reply:
if we dont actualy move and MOVED dont called we dont create trails in old turf that saved in PRE_MOVE